### PR TITLE
add support for summarize endpoint

### PR DESCRIPTION
**Pull request by daulet**
Add support for summarize endpoint. Tested against local deployment.

### CODE REVIEW

**Comment by choicallum** [comment]
lgtm right now but gonna need to add other fields in response / request

File: cohere/summarize.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from cohere.response import CohereObject
+
+
+class Summary(CohereObject):
+    def __init__(self, text: str) -> None:
+        self.text = text
```
Comment (choicallum) [inline cohere/summarize.py:6]: initially looked a bit odd to me (shouldn't this field be summary?) but summary.summary also doesn't make sense and this is consistent with generate so fine by me

File: cohere/client.py [hunk 2]
```diff
@@ -232,11 +232,28 @@ def unstable_extract(self, examples: List[ExtractExample], texts: List[str]) ->
...
+        self, 
+        text: str, 
+        length: str = "normal", 
+        useBaseline: bool = False,
+        max_tokens: int = 120,
+        temperature: float = 1.0,
+        k: int = 1,
+        p: float = 1,
+        frequency_penalty: float = 0.0,
+        presence_penalty: float = 0.0
```
Comment (daulet) [inline cohere/client.py:245]: @choicallum don't duplicate defaults here. If we change defaults in the service they'd be ignored by SDK users

**Comment by aidangomez** [comment]
@daulet @choicallum do you want to update this for merger or close for now?

**Comment by choicallum** [comment]
> @daulet @choicallum do you want to update this for merger or close for now?

Summarize is a cohere-only endpoint for now, part of why we wanted to push towards having an internal repo for development. We can probably close for now, but definitely no merge yet.

**Pull request closed without merging**